File: backend/internal/model/room.go
package model

import (
	"time"

	"gorm.io/gorm"
)

// Room 房间表
type Room struct {
	ID        uint           `gorm:"primaryKey" json:"id"`
	OwnerID   uint           `gorm:"index" json:"owner_id"`
	Owner     User           `gorm:"foreignKey:OwnerID" json:"owner"`
	Title     string         `gorm:"type:varchar(255);not null" json:"title"`
	Cover     string         `gorm:"type:varchar(255)" json:"cover"`
	CreatedAt time.Time      `json:"created_at"`
	UpdatedAt time.Time      `json:"updated_at"`
	DeletedAt gorm.DeletedAt `gorm:"index" json:"-"`
}

File: frontend/app/src/main/java/love/moonc/sparklink/data/remote/model/entity/Room.kt
package love.moonc.sparklink.data.remote.model.entity

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

@Serializable
data class Room(
    @SerialName("id") val id: Long,
    @SerialName("owner_id") val ownerId: Long,
    @SerialName("owner") val owner: User,
    @SerialName("title") val title: String,
    @SerialName("cover") val cover: String?,
    @SerialName("created_at") val createdAt: String,
    @SerialName("updated_at") val updatedAt: String
)